granulate_utils/metadata/databricks_client.py

```diff
@@ -11,21 +11,23 @@
 
 import requests
 
+from granulate_utils.exceptions import DatabricksJobNameDiscoverException
+from granulate_utils.metadata import Metadata
```

Comment:
No need to use this `Metadata` object, I don't even recall why we used it and I'd like to get it removed. Hiding the real type doesn't help and since it is so opaque, it's being reused for different purposes than was meant, as you're doing here :/ 

Use `Dict` directly.